lib/lambda_cache.ex

```diff
@@ -3,24 +3,24 @@ defmodule LambdaCache do
   Zero-arity function polling cache.
   """
 
-  @behaviour refresh :: any()
-  @behaviour interval :: non_neg_integer()
+  @callback refresh :: any()
+  @callback interval :: non_neg_integer()
 
   defmacro __using__(options) do
     quote location: :keep do
       @behaviour LambdaCache
 
-      def child_spec(args) do
+      use GenServer
+
+      def child_spec(args \\ nil) do
```

Comment:
I'm not sure about `nil` here either. What's the use case? `child_spec/0` typically presumes `[]`, not nil.

Reply:
i.e. even if you pass `__MODULE__` as hypervisor child (i.e. atom instead of a tuple), it will actually call `child_spec/1` with `[]` arg, not `child_spec/0`. Nevertheless, `\\ []` is the correct default here if you want one.